android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -361,6 +361,14 @@ class QuestionnaireActivity : BaseMultiLanguageActivity() {
           lifecycleScope.launch {
             retrieveQuestionnaireResponse()?.let { questionnaireResponse ->
               viewModel.saveDraftQuestionnaire(questionnaireResponse, questionnaireConfig)
+              setResult(
+                Activity.RESULT_OK,
+                Intent().apply {
+                  putExtra(QUESTIONNAIRE_RESPONSE, questionnaireResponse as Serializable)
+                  putExtra(QUESTIONNAIRE_CONFIG, questionnaireConfig as Parcelable)
+                  putExtra(ON_RESULT_TYPE, ActivityOnResultType.QUESTIONNAIRE.name)
+                },
+              )
```

Comment:
@FikriMilano Is there a reason this is also done here? The `setResult` function is called in the `handleQuestionnaireSubmission` does it mean that line of code is never reached?

Reply:
I see. So, the plan is to set the result so that you can set the task's status as in-progress :+1:.

Reply:
you had the right guess.
I'll do that then, thanks for the understanding